ospfd/ospfd.c

```diff
@@ -117,12 +116,12 @@ void ospf_router_id_update(struct ospf *ospf)
 	else if (ospf->router_id.s_addr != 0)
 		router_id = ospf->router_id;
 	else
-		router_id = router_id_zebra;
+		router_id = ospf->router_id_zebra;
 
 	if (IS_DEBUG_OSPF_EVENT)
 		zlog_debug("Router-ID[OLD:%s]: Update to %s",
 			   inet_ntoa(ospf->router_id),
-			   inet_ntoa(router_id_old));
+			   inet_ntoa(router_id));
```

Comment:
We are supposed to log the old router id here. Why are you changing it to router_id ?